biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java

```diff
@@ -3690,4 +3692,9 @@ private Boolean assignable0(Clazz annoServiceClazz, Clazz inferredServiceClazz)
 		return assignable0(findClass(zuper), inferredServiceClazz);
 	}
 
+	private void addDefinedContracts() {
+		Parameters contractEmulations = getMergedParameters(Constants.DEFINE_CONTRACT);
+		contracts.collectContracts("emulated", contractEmulations);
```

Comment:
Still using emulated here in the string and variable name. The `from` here is bnd configuration vs. a jar file.

Reply:
fixing